smr.js

```diff
@@ -31,17 +28,17 @@ MatrixProduct.prototype.push = MatrixProduct.prototype.addRowAndColumn
 
 function Regression(options) {
 
-  if(!options)
-    throw new Error('missing options')
+  // if(!options)
+  //   throw new Error('missing options')
 
-  if(!('numX' in options))
-    throw new Error(
-      'you must give the width of the X dimension as the property numX')
+  // if(!('numX' in options))
+  //   throw new Error(
+  //     'you must give the width of the X dimension as the property numX')
 
 
-  if(!('numY' in options))
-    throw new Error(
-      'you must give the width of the X dimension as the property numY')
+  // if(!('numY' in options))
+  //   throw new Error(
+  //     'you must give the width of the X dimension as the property numY')
```

Comment:
I'm surprised those particular checks are causing performance issues, given that they only happen during the original construction of the regressor... is that a use-case one should care about?

Since that piece of code will fail-fast without the checks (if any of the conditions aren't true, an error will be thrown immediately), the additional checks serve only two purposes:

  1. To provide more detailed feedback to a new user
  2. To ensure that `numX` and `numY` aren't grabbed from farther up the prototype chain, which might be considered undesirable.

I think the consideration here should be whether or not the marginal performance increase to what seems to me to be a fairly strange use-case is worth adding to on-boarding time for new users. Up to you, of course.

I'd like to see actual numbers and code demonstrating the slowdown created by these checks...